odk2odm/odk_api.py

```diff
@@ -91,17 +111,89 @@ def attachment(base_url, aut, projectId, formId, instanceId, filename):
         f'{instanceId}/attachments/{filename}'
     return requests.get(url, auth = aut)
 
+# POST 
 
 def create_project(base_url, aut, project_name):
     """Create a new project on an ODK Central server"""
     url = f'{base_url}/v1/projects'
     return requests.post(url, auth = aut, json = {'name': project_name})
 
+def create_app_user(base_url, aut, projectId, app_user_name = 'Surveyor'):
+    """Create a new project on an ODK Central server"""
+    url = f'{base_url}/v1/projects/{projectId}/app-users'
+    return requests.post(url, auth = aut, json = {'displayName': app_user_name})
+
+
+#Not sure if this is needed
+def create_public_link(base_url, aut, projectId, formId, linkName):
+    """Give access to anybody to submit a form"""
+    url = f'{base_url}/v1/projects/{projectId}/forms/{formId}/public-links'
+    return requests.post(url, auth = aut, json = {'displayName': linkName}) 
+
+
+def give_access_app_users(base_url, aut, projectId):
+    """Give all the app-users in the project access to all the forms in that project"""
+    url = f'{base_url}/v1/projects/{projectId}/forms'
+    forms = requests.get(url, auth = aut).json()
+
+    for form in forms:
+        formId = form['xmlFormId']
+        url = f'{base_url}/v1/projects/{projectId}/app-users'
+        app_users = requests.get(url, auth = aut).json()
+        for user in app_users:
+            actorId = user['id']        
+            roleId = 2
+            url = f'{base_url}/v1/projects/{projectId}/forms/{formId}/assignments/{roleId}/{actorId}' 
+            requests.post(url, auth = aut)  
+
 
 def delete_project(base_url, aut, project_id):
     """Permanently delete project from an ODK Central server. Probably don't."""
     url = f'{base_url}/v1/projects/{project_id}'
     return requests.delete(url, auth = aut)
 
+###### Editing
+def create_form(base_url, aut, projectId, path2Form):
+    """Create a new form on an ODK Central server"""
+    base_name = os.path.basename(path2Form)
+    file_name = os.path.splitext(base_name)[0]
+    form_file = open(path2Form, 'rb')
+    #sheet = form_file.active
+
+    headers = {
+    'Content-Type': 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    f'X-XlsForm-FormId-Fallback': file_name
+    }
+    url = f'{base_url}/v1/projects/{projectId}/forms?ignoreWarnings=true&publish=true'
+    values = form_file
+
```

Comment:
ignoreWarnings and publish are keywords that could be configured by a user. Now they are fixed entirely. Can you make these a separate data field added to the post request?